### DISCUSSION TITLE
T-parameter definition clarification

### DISCUSSION THREAD

**Discussion by crr9542** [scikit-rf/scikit-rf#1088] [Q&A] [answered]
Been reading the documentation as much as possible and was just confused since it's not necessarily outlined: which of the below definitions for T-parameters of a 2-port network is being used? The Wikipedia as linked in the docs contains both the (b1 over a!) = t * (a2 over b2) as well as the other definition- (a1 over b1) = t * .... 

Hopefully a simple question (and sorry, would have been a lot easier if my photos would link!)

**Comment by jhillairet** [comment, marked as answer]
The T-parameters in scikit-rf correspond to the following definition:

![image](https://github.com/scikit-rf/scikit-rf/assets/4642848/478c4dca-8cdc-444c-a9e6-eefcd51fa07c)

More references:
https://doi.org/10.1109/TMTT.2008.2005873
Rizzi, Microwave Engineering

Opposite definition:
https://doi.org/10.1109/22.275248
https://doi.org/10.1109/INMMIC.2018.8430023

**Comment by ZiadHatab** [comment]
I know this is old, but one thing I will add here is that it all depends on what "a" and "b" incident and reflect from. The best way I found is to stay consistent in the direction of the arrow. For example, all arrows going from left to right are incident waves, and arrows going the other way are reflected. The key point is that you want the elements of the wave-vectors to be in the same arrow direction.

In the example below, I used reflected wave as top, but you can swap "a" and "b" and you will get the opposite definition for T-parameters. It is always good to imagine there is a DUT terminating the network.

![image](https://github.com/user-attachments/assets/3d7a930d-e4bf-46de-a50d-6f7c61a81d85)
